### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by Tahirbhalli**
> build with
  > html
  > bootstrap
  > javascript

### CODE REVIEW

File: script/data.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  const td = document.getElementById('books');
+  const row = document.createElement('tr');
+  row.appendChild(create(book.title));
+  row.appendChild(create(book.author));
+  row.appendChild(create(book.price));
+
+
+  row.appendChild(createbutton(book));
+  td.appendChild(row);
+}
```
Comment (bellom) [inline script/data.js]: Let's separate the DOM from application logic, you can create another file for DOM.

image

https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59623451/

File: index.html [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                    <div class="form-group">
+                      <input type="text" id="author" placeholder="Enter name of author">
+                    </div>
+                    <div class="form-group">
+                      <input type="number" id="price" min=0 placeholder="Enter Price">
+                    </div>
+                    <div class="form-group text-center">
+                      <button type="submit" class="btn btn-primary" onclick="add(document.getElementById('title').value,document.getElementById('author').value,document.getElementById('price').value)">Add</button>
+                    </div>
+                </form>
```
Comment (bellom) [inline index.html]: Kindly validate your form, users should not be able to add an empty form.

File: script/data.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+// eslint-disable-next-line no-unused-vars
+function getbooks() {
+  if (localStorage.getItem('books') === null) {
+    books = [new Book('javascript', 'Ghosh', 10, 'Read'), new Book('css', 'Ghosh', 10, 'Read'), new Book('html', 'Ghosh', 10, 'Read')];
```
Comment (bellom) [inline script/data.js:23]: We need to add a `remove` button to each book,  you can easily make use of `data-atrr`
image
https://www.theodinproject.com/courses/javascript/lessons/library

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

File: script/data.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
+class Book {
```
Comment (bellom) [inline script/data.js]: we are missing the render function, I can see your `getBook` function, but one of the project requirement is to have a render function that loops through the book array and displays each book on the page.
Kindly create the render function as we both know all other students follow this project guideline too. 👍 
image
https://www.theodinproject.com/courses/javascript/lessons/library

**Review by shloch — CHANGES_REQUESTED** [review]
## Hello @Tahirbhalli , 

### changes required]

please control the form like former TSE said, control submission of empty forms
book

let's do this

**Review by maelfosso — APPROVED** [review]
# Congratulations :red_circle: 

Thank  you for applying the suggested changes. It's working now like a charm :smile: 
You can merge it :+1: 

Good continuation :handshake: 

Mael FOSSO

**Pull request merged**